### ISSUE TITLE
App doesn't start if the network port is already taken

### ISSUE THREAD

**Issue by manuq** [flathub/org.learningequality.Kolibri#16]
If the port is taken by another process, this error happens:
![Captura de pantalla de 2020-09-16 10-08-21](https://user-images.githubusercontent.com/83944/93341825-f314b680-f804-11ea-9b74-a03bc9c9d8e6.png)

Traceback:

```
$ flatpak run org.learningequality.Kolibri//stable
INFO     Running Kolibri with the following settings: kolibri_gnome.kolibri_settings
ERROR    Port 8080 is occupied.
Please check that you do not have other processes running on this port and try again.

Traceback (most recent call last):
  File "/app/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri-flatpak-utils==0.0.0', 'console_scripts', 'kolibri')()
  File "/app/lib/python3.7/site-packages/kolibri_flatpak/kolibri_wrapper.py", line 119, in main
    return application.run()
  File "/app/lib/python3.7/site-packages/kolibri_flatpak/kolibri_wrapper.py", line 40, in run
    return self.__run_kolibri()
  File "/app/lib/python3.7/site-packages/kolibri_flatpak/kolibri_wrapper.py", line 104, in __run_kolibri
    result = subprocess.run([KOLIBRI, *args], env=self.__kolibri_env, check=True)
  File "/usr/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/app/libexec/kolibri', 'start', '--foreground']' returned non-zero exit status 1.
WARNING:kolibri_gnome.desktop_launcher.application:Kolibri service has died
```

**Comment by dylanmccall** [comment]
Thanks for filing this one! I've noted it a few times but never really got around to writing it down :) What we should do here is, in [Kolibri](https://github.com/learningequality/kolibri), allow Kolibri to start with a randomized port number (`KOLIBRI_HTTP_PORT=0`). We need that to support multiple Kolibri channel applications running their own Kolibri servers, as well, and it would really help with stability here. I don't think there's actually too much that needs to change there, since cherrypy already does the right thing. It's just that Kolibri's surrounding startup code needs to get the port number cherrypy is using rather than expecting the environment variable to be the same.

**Comment by dylanmccall** [comment]
This was fixed with pull request #17.
